components/authentication/azure/src/main/java/com/microsoft/kiota/authentication/AzureIdentityAccessTokenProvider.java

```diff
@@ -113,8 +106,13 @@ public CompletableFuture<String> getAuthorizationToken(@Nonnull final URI uri, @
             if(decodedClaim != null && !decodedClaim.isEmpty()) {
                 context.setClaims(decodedClaim);
             }
-            return this.creds.getToken(context).toFuture().thenApply(r -> r.getToken());
-        } finally {
+            return this.creds.getToken(context).toFuture().join().getToken();
```

Comment:
Any reason to not use the sync overload?
```suggestion
            return this.creds.getTokenSync(context).getToken();
```

Reply:
Good catch